fix : Show savings account status with reference to issue #44

sagar15795 opened the pull request:
Changes
- Added circular_background xml for layered image 
- Added imageview in row_saving_account.xml
- Added circularImageView class with attr.xml
- Checked multiple status conditions in SavingAccountsListAdapter and show accordingly
- Implemented getter and setter methods for status and other fields

sagar15795 commented:
Now Saving Account list UI look like

![image](https://cloud.githubusercontent.com/assets/5468111/19868865/f344ae4e-9fcf-11e6-8959-cb59987409fc.png)

therajanmaurya commented:
@sagar15795 Squash the commits to single commits before making PR.
and Please follow this [Commit Style Guidelines](https://github.com/openMF/self-service-app/wiki/Commit-Style-Guide)
and [Pull request guidelines](https://github.com/openMF/self-service-app/blob/development/.github/CONTRIBUTING.md)

sagar15795 commented:
@therajanmaurya  I have squashed the commits. Please review it now.

therajanmaurya commented on app/src/main/res/layout/row_saving_account.xml:
```diff
@@ -9,6 +9,14 @@
     android:paddingRight="@dimen/row_padding_right"
 >
+    <org.mifos.selfserviceapp.utils.CircularImageView
+        android:id="@+id/view_status_indicator"
```
Id should be iv_status_indicator not view_status_indicator.

therajanmaurya commented on app/src/main/res/values/dimens.xml:
```diff
@@ -1,11 +1,14 @@
     <!-- Default screen margins, per the Android Design guidelines. -->
 
+    <!--Dimen values for width-->
+    <dimen name="row_status_width">25dp</dimen>
```
ambiguous naming, `row_status_width` can be `account_status_width`.

therajanmaurya commented on app/src/main/res/values/dimens.xml:
```diff
@@ -1,11 +1,14 @@
     <dimen name="toolbar_height">32dp</dimen>
     <dimen name="tab_layout_height">28dp</dimen>
     <dimen name="login_height">72dp</dimen>
+    <dimen name="row_status_height">25dp</dimen>
```
ambiguous naming, `row_status_height` can be `account_status_height`
and put both  `account_status_height` and `account_status_width` simultaneously like this.

``` java
 <!-- Dimen values for Account Status-->
<dimen name="account_status_height">25dp</dimen>
<dimen name="account_status_width">25dp</dimen>
```

therajanmaurya commented on app/src/main/res/values/dimens.xml:
```diff
@@ -27,6 +30,7 @@
     <dimen name="saving_accounts_text_view_margin_left">50dp</dimen>
     <dimen name="loan_accounts_text_view_margin_left">50dp</dimen>
     <dimen name="client_charge_text_view_margin_left">20dp</dimen>
+    <dimen name="row_status_margin_end">8dp</dimen>
```
According to android Material Guideline, 8dp is the standard margin.
So, `row_status_margin_end` can be `default_margin` and also remove 

``` java
<dimen name="login_text_input_margin_top">8dp</dimen>
<dimen name="login_text_input_margin_bottom">8dp</dimen>
```

and use `default_margin`.

therajanmaurya commented on app/src/main/java/org/mifos/selfserviceapp/ui/adapters/SavingAccountsListAdapter.java:
```diff
@@ -63,12 +91,29 @@ public int getItemCount() {
         TextView tv_savingAccountProductName;
 
+        @BindView(R.id.view_status_indicator)
+        CircularImageView imageview_status_indicator;
```
`imageview_status_indicator` can be `iv_status_indicator`

therajanmaurya commented on app/src/main/java/org/mifos/selfserviceapp/ui/adapters/SavingAccountsListAdapter.java:
```diff
@@ -63,12 +91,29 @@ public int getItemCount() {
         return savingAccountsList.size();
     }
 
+    private LayerDrawable setCircularBackground(int colorId) {
```
Remove the one line gap between lines.
should be like this 

``` java
 private LayerDrawable setCircularBackground(int colorId) {
        Drawable color = new ColorDrawable(ContextCompat.getColor(context, colorId));
        Drawable image = ContextCompat.getDrawable(context, R.drawable.circular_background);
        LayerDrawable ld = new LayerDrawable(new Drawable[]{image, color});
        return ld;
    }
```

therajanmaurya commented on app/src/main/java/org/mifos/selfserviceapp/ui/adapters/SavingAccountsListAdapter.java:
```diff
@@ -54,6 +59,29 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
             ((ViewHolder) holder).tv_savingAccountProductName.setText(
                     savingAccount.getProductName());
 
+
```
I have looked the adapter and found `private final LayoutInflater layoutInflater` is unused remove this and 

``` java
((ViewHolder) holder).tv_clientSavingAccountNumber.setText(
                    savingAccount.getAccountNo().toString());
```

here Account No already a string, So toString() is redundant , remove toString().
and 
`if (holder instanceof RecyclerView.ViewHolder)` this check is redundant because
`RecyclerView.ViewHolder` is RecyclerView abstract standard class but here we have check holder is the instance of the SavingAccountsListAdapter.ViewHolder or not, So remove `RecyclerView.ViewHolder` to SavingAccountsListAdapter's ViewHolder.

sagar15795 commented:
@therajanmaurya  I have added all suggestions by you.
Please merge it. 
Thanks

therajanmaurya commented on app/src/main/res/values/dimens.xml:
```diff
@@ -1,6 +1,8 @@
 <resources>
     <!-- Default screen margins, per the Android Design guidelines. -->
 
+    <!--Dimen values for width-->
```
merge this comment and below comment.

sagar15795 commented:
@therajanmaurya I have removed the redundant comment.

sagar15795 committed:
fix: Show savings account status
Added circular_background xml for layered image
Added imageview in row_saving_account.xml
Added circularImageView class with attr.xml
Checked multiple status conditions in SavingAccountsListAdapter and show accordingly
Implemented getter and setter methods for status and other fields
